core/embed/extmod/modtrezorcrypto/modtrezorcrypto-optiga.h

```diff
@@ -0,0 +1,113 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#if USE_OPTIGA
+
+#include "py/objstr.h"
+
+#include "optiga.h"
+
+/// package: trezorcrypto.optiga
+
+#define MAX_DER_SIGNATURE_SIZE 72
+
+/// def get_certificate(cert_index: int) -> bytes:
+///     """
+///     Return the certificate stored at the given index.
+///     """
+STATIC mp_obj_t mod_trezorcrypto_optiga_get_certificate(mp_obj_t cert_index) {
+  mp_int_t idx = mp_obj_get_int(cert_index);
+  if (idx < 0 || idx > 3) {
+    mp_raise_ValueError("Invalid index.");
+  }
+
+  size_t cert_size = 0;
+  if (!optiga_cert_size(idx, &cert_size)) {
+    mp_raise_ValueError("Failed to get certificate size.");
+  }
+
+  vstr_t cert = {0};
+  vstr_init_len(&cert, cert_size);
+  if (!optiga_read_cert(idx, (uint8_t *)cert.buf, cert.alloc, &cert_size)) {
+    vstr_clear(&cert);
+    mp_raise_ValueError("Failed to read certificate.");
+  }
+
+  cert.len = cert_size;
+  return mp_obj_new_str_from_vstr(&mp_type_bytes, &cert);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(mod_trezorcrypto_optiga_get_certificate_obj,
+                                 mod_trezorcrypto_optiga_get_certificate);
+
+/// def sign(
+///     key_index: int,
+///     digest: bytes,
+/// ) -> bytes:
+///     """
+///     Uses the private key at key_index to produce a DER-encoded signature of
+///     the digest.
+///     """
+STATIC mp_obj_t mod_trezorcrypto_optiga_sign(mp_obj_t key_index,
+                                             mp_obj_t digest) {
+  mp_int_t idx = mp_obj_get_int(key_index);
+  if (idx < 0 || idx > 3) {
+    mp_raise_ValueError("Invalid index.");
+  }
+
+  mp_buffer_info_t dig = {0};
+  mp_get_buffer_raise(digest, &dig, MP_BUFFER_READ);
+  if (dig.len != 32) {
+    mp_raise_ValueError("Invalid length of digest.");
+  }
+
+  vstr_t sig = {0};
+  vstr_init_len(&sig, MAX_DER_SIGNATURE_SIZE);
+  size_t sig_size = 0;
+  int ret = optiga_sign(idx, (const uint8_t *)dig.buf, dig.len,
+                        ((uint8_t *)sig.buf), sig.alloc, &sig_size);
+  if (ret != 0) {
+    vstr_clear(&sig);
+    if (ret == OPTIGA_ERR_ACCESS_COND_NOT_SAT) {
+      mp_raise_msg(&mp_type_RuntimeError, "Signing inaccessible.");
+    } else {
+      mp_raise_msg(&mp_type_RuntimeError, "Signing failed.");
+    }
+  }
+
+  sig.len = sig_size;
+  return mp_obj_new_str_from_vstr(&mp_type_bytes, &sig);
+}
+
+STATIC MP_DEFINE_CONST_FUN_OBJ_2(mod_trezorcrypto_optiga_sign_obj,
+                                 mod_trezorcrypto_optiga_sign);
+
+STATIC const mp_rom_map_elem_t mod_trezorcrypto_optiga_globals_table[] = {
+    {MP_ROM_QSTR(MP_QSTR___name__), MP_ROM_QSTR(MP_QSTR_optiga)},
+    {MP_ROM_QSTR(MP_QSTR_get_certificate),
+     MP_ROM_PTR(&mod_trezorcrypto_optiga_get_certificate_obj)},
+    {MP_ROM_QSTR(MP_QSTR_sign), MP_ROM_PTR(&mod_trezorcrypto_optiga_sign_obj)}};
```

Comment:
we should export something like `DEVICE_CERTIFICATE_SLOT` so that we can use it in `optiga.sign(optiga.DEVICE_CERTIFICATE_SLOT, ...)`

Reply:
https://github.com/trezor/trezor-firmware/pull/3255/commits/e23bfb02537b47ac8a23419d53ada39a7566eafc https://github.com/trezor/trezor-firmware/pull/3255/commits/b2badfaa3e1d9aa66b1c5f5dfcc65344d3f81b2e